supabase/migrations/50_controllers.sql

```diff
@@ -0,0 +1,127 @@
+begin;
+
+
+create type flow_type as enum (
+  -- These correspond 1:1 with catalog_spec_type.
+  'capture',
+  'collection',
+  'materialization',
+  'test',
+  -- These do not
+  'source_capture'
+);
+comment on type flow_type is
+  'Represents the type of a dependency of one spec on another. This enum is a
+  strict superset of catalog_spec_type, for historical reasons.';
+
+-- This works because the `flow_type` enum is a superset of the `catalog_spec_type` enum.
+-- This approach was taken from:
+-- https://www.munderwood.ca/index.php/2015/05/28/altering-postgresql-columns-from-one-enum-to-another/
+alter table live_spec_flows
+alter column flow_type set data type flow_type
+using flow_type::text::flow_type;
+
+-- We now allow live specs to be deleted even though they're still referenced by other specs.
```

Comment:
Originally this was here because controllers would hard delete `live_specs` and then call `notify_dependents`, so the `live_spec_flows` were still needed in order to identify the dependent specs to notify. But now, the deletion of live specs happens _after_ the call to `notify_dependents`, and so I would think that we can keep these constraints and add `on delete cascade`. I'll give it a go

Reply:
Following up: the constraints have been re-added along with `on delete cascade`